src/sanitize/sanitizeNumbers.ts

```diff
@@ -1,94 +1,140 @@
-/* eslint-disable @typescript-eslint/no-unsafe-assignment */
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
-/* eslint-disable @typescript-eslint/no-unsafe-call */
-/* eslint-disable @typescript-eslint/no-unsafe-return */
 import {
 	BTreeSet,
 	Compact,
 	Enum,
-	HashMap,
+	Option,
 	Set as CodecSet,
 	Struct,
 } from '@polkadot/types';
+import AbstractArray from '@polkadot/types/codec/AbstractArray';
 import AbstractInt from '@polkadot/types/codec/AbstractInt';
-import Option from '@polkadot/types/codec/Option';
-import { AnyJson, Codec } from '@polkadot/types/types';
+import CodecMap from '@polkadot/types/codec/Map';
+import StructAny from '@polkadot/types/codec/StructAny';
+import { isObject } from '@polkadot/util';
 import * as BN from 'bn.js';
+import { InternalServerError } from 'http-errors';
+
+import {
+	AnyJson,
+	Codec,
+	isAnyJson,
+	isCodec,
+	isToJSONable,
+} from '../types/polkadot-js';
 
 /**
- * A sanitizer for arbitrary data that's going to be
- * stringified to JSON. We find all instances of `AbstractInt`,
- * which is using bn.js as backend, and forcibly serialize it
- * to a decimal string.
+ * Forcibly serialize all instances of AbstractInt to base 10. With Codec
+ * based types we can provide a strong guarantee that the output will be of AnyJson
  *
- * @param data
+ * @param value a type that implements polkadot-js Codec
  */
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
-export function sanitizeNumbers(data: any): any {
-	if (data instanceof Option) {
-		return data.isSome ? sanitizeNumbers(data.unwrap()) : null;
+function sanitizeCodec(value: Codec): AnyJson {
+	// If objects have an overlapping prototype chain
+	// we check lower down the chain first. More specific before less specific.
+
+	if (value instanceof Option) {
+		return value.isSome ? sanitizeNumbers(value.unwrap()) : null;
 	}
 
-	if (data instanceof Compact) {
-		return sanitizeNumbers(data.unwrap());
+	if (value instanceof Compact) {
+		return sanitizeNumbers(value.unwrap());
 	}
 
-	if (data instanceof Enum) {
-		if (data.isBasic) {
-			return data.toJSON();
+	// // For the sake of API consistency this is staying out for now
+	// if (value instanceof Result) {
+	// 	return value.isOk
+	// 		? sanitizeNumbers(value.asOk)
+	// 		: sanitizeNumbers(value.asError);
+	// }
```

Comment:
I would avoid checking in comments like this. Consider leaving a `// TODO` instead.